exec/cnex/format.c

```diff
@@ -0,0 +1,157 @@
+#include "format.h"
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#include "exec.h"
+#include "number.h"
+#include "nstring.h"
+
+BOOL parse(TString *fmt, Spec *spec)
+{
+    spec->fill = ' ';
+    spec->align = 0;
+    spec->alternate = FALSE;
+    spec->sign = 0;
+    spec->width = -1;
+    spec->thousands = FALSE;
+    spec->precision = -1;
+    spec->type = 0;
+    if (string_isEmpty(fmt)) {
+        return TRUE;
+    }
+    size_t i = 0;
+    if (fmt->data[i] == '<' || fmt->data[i] == '>' || fmt->data[i] == '=' || fmt->data[i] == '^') {
+        spec->align = fmt->data[i];
+        ++i;
+    }
+    if (fmt->data[i] == '+' || fmt->data[i] == '-' || fmt->data[i] == ' ') {
+        spec->sign = fmt->data[i];
+        ++i;
+    }
+    if (fmt->data[i] == '0') {
+        spec->fill = '0';
+        if (spec->align == 0) {
+            spec->align = '=';
+        }
+        ++i;
+    }
+    if (isdigit(fmt->data[i])) {
+        size_t end = 0;
+        char *s = string_asCString(string_subString(fmt, i, fmt->length-1));
+        spec->width = atoi(s);
+        free(s);
+        i += end;
+    }
+    if (fmt->data[i] == ',') {
+        spec->thousands = TRUE;
+        ++i;
+    }
+    if (fmt->data[i] == '.') {
+        ++i;
+        size_t end = 0;
+        char *s = string_asCString(string_subString(fmt, i, fmt->length-1));
+        spec->precision = atoi(s);
+        free(s);
+        i += end;
+    }
+    if (i >= fmt->length - 1) {
+        spec->type = '\0';
+    } else {
+        spec->type = fmt->data[i];
+    }
+    switch (spec->type) {
+        case 0:
+        case 's':
+            if (spec->sign || spec->alternate || spec->align == '=') {
+                return FALSE;
+            }
+            break;
+        case 'd':
+        case 'x':
+            return TRUE;
+        default:
+            return FALSE;
+    }
+    return TRUE;
+}
+
+static TString *fillstr(Spec *spec, int n)
+{
+    TString *r = string_newString();
+    if (spec->fill < 128) {
```

Comment:
I know why this `< 128` test is here, because I wrote the code you're cloning, but for *your* code it's completely unclear why you are doing this at all. And you don't even need to at this point.

Reply:
Removed entire block.